### PR TITLE
Added base popups

### PR DESCRIPTION
**Pull request by MishaAfanasiuk**
Added base popups
Created confirmation, information and error popups

https://trello.com/c/hLaYehaO

### CODE REVIEW

File: src/components/shared/BasePopup/index.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  body,
+  popupClassName,
+  footer
+}) => {
+
+  const backgroundHandler = (e) => {
+    if (e.currentTarget === e.target) {
+      onClose();
+    }
+  };
```
Comment (MishaHnatyshyn) [inline src/components/shared/BasePopup/index.js]: Can we use useCallback or useMemo here?

File: src/components/shared/BasePopup/index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { CSSTransition } from 'react-transition-group';
+import PopupButtons from './components/PopupButtons';
+import styles from './base.popup.module.scss';
+import './styles.scss';
```
Comment (MishaHnatyshyn) [inline src/components/shared/BasePopup/index.js:6]: What is this? We use scss modules for styles

File: src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  cursor: pointer;
+  outline: none;
+  transition: all .15s linear;
+
+  font: $text-button-bold-small;
+  color: $black;
+
+  &:hover {
+    box-shadow: 0px 5px 13px -2px rgba(0,0,0,0.32);
```
Comment (MishaHnatyshyn) [inline src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss:18]: move to variables ?

File: src/components/shared/BasePopup/base.popup.module.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+.popup {
+  max-width: 410px;
+  min-width: 320px;
+  position: relative;
+
+  padding: 25px;
+
+  background-color: $white;
+  box-shadow: 0px 0px 19px 3px rgba(0,0,0,0.37);
```
Comment (MishaHnatyshyn) [inline src/components/shared/BasePopup/base.popup.module.scss:44]: move to variables?

File: src/components/shared/BasePopup/base.popup.module.scss [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+.popupWrapper {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: absolute;
+  height: 100vh;
+  width: 100%;
+
+  z-index: 99;
```
Comment (MishaHnatyshyn) [inline src/components/shared/BasePopup/base.popup.module.scss]: move to variables?

File: src/components/PopupExamples.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+// TODO: Remove this file asap )
```
Comment (MishaHnatyshyn) [inline src/components/PopupExamples.js]: Why did you add it ))?

**Review by MishaHnatyshyn — CHANGES_REQUESTED** [review]

File: src/components/PopupExamples.js (hunk 6 above)
Comment (MishaAfanasiuk) [reply src/components/PopupExamples.js]: Read next line ))

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 5 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss]: \-

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 4 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss:44]: \-

File: src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss (hunk 3 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/components/PopupButton/popupButton.module.scss:18]: \-

File: src/components/shared/BasePopup/index.js (hunk 2 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/index.js:6]: Scss module didn't work with react-transition for me

File: src/components/shared/BasePopup/index.js (hunk 1 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/index.js]: \+

File: src/components/PopupExamples.js (hunk 6 above)
Comment (MishaHnatyshyn) [reply src/components/PopupExamples.js]: I see, but why did you add something in order to delete in the future ))?

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 5 above)
Comment (MishaHnatyshyn) [reply src/components/shared/BasePopup/base.popup.module.scss]: why ?

Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss]: Why move it to variables?

File: src/components/PopupExamples.js (hunk 6 above)
Comment (MishaAfanasiuk) [reply src/components/PopupExamples.js]: Because it is useless ))

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 5 above)
Comment (MishaHnatyshyn) [reply src/components/shared/BasePopup/base.popup.module.scss]: I believe it to be useful to store all z-indexes in one place in order not to search in the future "Why my new custom popup is below this one"

File: src/components/PopupExamples.js (hunk 6 above)
Comment (MishaHnatyshyn) [reply src/components/PopupExamples.js]: if it is useless so..... ??

https://ru.wikipedia.org/wiki/YAGNI

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 5 above)
Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss]: Show me an example

Comment (MishaHnatyshyn) [reply src/components/shared/BasePopup/base.popup.module.scss]: I want to create some toast notification that should be displayed above this popup. Why should I search for this popup component and its styles file in order to find its z-index, instead of just look at the varaibles file and create a new variable with greater z-index or just use the same z-index as a variable.

Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss]: How are you going to find it in variables?

And why not to check for z-index in chrome console (two clicks), and add a higher index.

Comment (MishaAfanasiuk) [reply src/components/shared/BasePopup/base.popup.module.scss]: Moved to variables

**Review by MishaHnatyshyn — APPROVED** [review]
